cmake/Dependencies.cmake

```diff
@@ -1263,11 +1263,11 @@ if (NOT INTERN_BUILD_MOBILE)
     MESSAGE(STATUS "Removing -DNDEBUG from compile flags")
     STRING(REGEX REPLACE "[-/]DNDEBUG" "" CMAKE_C_FLAGS "" ${CMAKE_C_FLAGS})
     STRING(REGEX REPLACE "[-/]DNDEBUG" "" CMAKE_C_FLAGS_DEBUG "" ${CMAKE_C_FLAGS_DEBUG})
-    STRING(REGEX REPLACE "[-/]DNDEBUG" "" CMAKE_C_FLAGS_RELEASE "" ${CMAKE_C_FLAGS_RELEASE})
     STRING(REGEX REPLACE "[-/]DNDEBUG" "" CMAKE_CXX_FLAGS "" ${CMAKE_CXX_FLAGS})
     STRING(REGEX REPLACE "[-/]DNDEBUG" "" CMAKE_CXX_FLAGS_DEBUG "" ${CMAKE_CXX_FLAGS_DEBUG})
-    STRING(REGEX REPLACE "[-/]DNDEBUG" "" CMAKE_CXX_FLAGS_RELEASE "" ${CMAKE_CXX_FLAGS_RELEASE})
   ENDIF()
+  STRING(APPEND CMAKE_C_FLAGS_RELEASE " -DNDEBUG")
+  STRING(APPEND CMAKE_CXX_FLAGS_RELEASE " -DNDEBUG")
```

Comment:
@peterjc123 Based on my reading of the docs, I think that this should take care of the multi-build-type generators since they should use `CMAKE_CXX_FLAGS_RELEASE`. Unfortunately, I work only on Linux and have no way to test this for MSVC or other generators. Perhaps I should stop working on this and leave it up to someone with more diverse environments?

Reply:
Actually, it's acceptable because we don't  use the MSBuild generator in our MSVC build. Alternatively, we use Ninja which is a single-build-type generator. Maybe we can open a new issue for this and skip that for now.

Reply:
@peterjc123 Just to make sure I understand you correctly, you are suggesting that I:
* revert the change of the `*_RELEASE` flags
* create a new issue to support MSBuild and possibly other multi-build-type generators. (If they are not being supported now, I suspect there will be more discrepancies than just NDEBUG).

Reply:
May I know what you are doing first? I'm a bit confused about the changes here. The code here is to remove `-DNDEBUG` from the compiler flags from both debug and release builds if `CMAKE_BUILD_TYPE` is `Release`.

Reply:
Sorry, changing `*_DEBUG` flags if build type is Release is indeed confusing. I will clean up that part and submit a clearer patch, even if only for the sake of discussion.